bin/tachyon-mount.sh

```diff
@@ -70,6 +70,64 @@ function mem_size_to_bytes() {
   esac
 }
 
+# Mac OS X HFS+ provisioning
+# Arguments:
+#   Requested filesystem size in bytes
+#   Sector size
+# Returns:
+#   Total sectors of HFS+ including estimated metadata zone size,
+#   guaranteeing available space of created HFS+ is not less than request size.
+# Notes:
+#   Metadata zone size is estimated by using specifications noted at
+#   http://dubeiko.com/development/FileSystems/HFSPLUS/tn1150.html#MetadataZone,
+#   which is slightly larger than actually metadata zone size created by hdiuitl command.
+function mac_hfs_provision_sectors() {
+  local request_size=$1
+  local sector_size=$2
+  local alloc_size=0
+  local n_sectors=0
+  local n_100g=0
+  local n_kb=0
+  local alloc_bitmap_size=0
+  local ext_overflow_file_size=0
+  local journal_file_size=0
+  local catalog_file_size=0
+  local hot_files_size=0
+  local quota_users_file_size=0
+  local quota_groups_file_size=0
+  local alloc_size=0
+
+  n_sectors=$((request_size / sector_size))
+  n_100g=`printf "%.0f" $(echo "scale=4; $request_size/107374182400 + 0.5" | bc)` # round up
+  n_gb=`printf "%.0f" $(echo "scale=4; $request_size/1073741824 + 0.5" | bc)` # round up
+  n_kb=`printf "%.0f" $(echo "scale=4; $request_size/1024 + 0.5" | bc)` #round up
+
+  # allocation bitmap file: one bit per sector
+  alloc_bitmap_size=$((n_sectors / 8))
+
+  # extends overflow file: 4MB, plus 4MB per 100GB
+  ext_overflow_file_size=$((4194304 * n_100g))
+
+  # journal file: 8MB, plus 8MB per 100GB
+  journal_file_size=$((8388608 * n_100g))
+
+  # catalog file: 10bytes per KB
+  catalog_file_size=$((10 * n_kb))
+
+  # hot files: 5bytes per KB
+  hot_files_size=$((5 * n_kb))
+
+  # quota users file and quota groups file
+  quota_users_file_size=$(((n_gb * 256 + 1) * 64))
+  quota_groups_file_size=$(((n_gb * 32 + 1) * 64))
+
+  metadata_size=$((alloc_bitmap_size + ext_overflow_file_size + journal_file_size + \
+    catalog_file_size + hot_file_size + quota_users_file_size + quota_groups_file_size))
+
+  alloc_size=$((request_size + metadata_size))
+  echo $((alloc_size / sector_size))
```

Comment:
By my estimation, it does not matter. Previous n_gb, n_kb, etc. calculation has been rounded up, which provides sufficient number for calculating metazone size. Also note that if round-down happens, it always rounds small values in bytes, such as 10, 5, 256, sector_size (512).